### PR TITLE
Frontend uses experiment name directly from ApiResourceReference

### PR DESCRIPTION
**Pull request by Bobgy**
Fixes https://github.com/kubeflow/pipelines/issues/1773

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1837)

### CODE REVIEW

File: frontend/src/lib/RunUtils.ts [hunk 1]
```diff
@@ -63,6 +63,25 @@ function getFirstExperimentReferenceId(run?: ApiRun | ApiJob): string | null {
...
 
+export interface ExperimentReferenceInfo {
+  name: string;
+  id: string;
+}
+
+function getFirstExperimentReferenceInfo(run?: ApiRun | ApiJob): ExperimentReferenceInfo | null {
+  if (run) {
+    const reference = getAllExperimentReferences(run)[0];
+    if (reference && reference.name && reference.key && reference.key.id) {
```
Comment (Bobgy) [inline frontend/src/lib/RunUtils.ts]: I am assuming backend should always provide the two together. Do you think it's a fair assumption?

**Comment by IronPan** [comment]
thanks @Bobgy It looks like @ajchili already had a fix for it https://github.com/kubeflow/pipelines/pull/1824

**Comment by Bobgy** [comment]
Some tests are failing because I didn't update mocked data. I will wait for discussion in #1824 to see whether I should keep working on this.

File: frontend/src/lib/RunUtils.ts (hunk 1 above)
Comment (ajchili) [reply frontend/src/lib/RunUtils.ts]: There is a possibility that the name is not provided when an older reference or lower version backend is used. Additionally, it is possible that the first experiment reference returned by `getAllExperimentReferences` is not a valid `ExperimentReferenceInfo` object. Due to this, all indexes of the array returned by `getAllExperimentReferences` should be checked to ensure that at least one is a valid `ExperimentReferenceInfo` object. This was discussed offline with @rileyjbauer and @IronPan.

**Comment by Bobgy** [comment]
After getting more knowledge about the context, I think #1824 alone is good enough. Closing this PR.

**Pull request closed without merging**